app/src/test/java/com/github/se/eventradar/ChatViewModelUnitTest.kt

```diff
@@ -0,0 +1,219 @@
+          accountStatus = "active",
+          eventsAttendeeList = mutableListOf("1", "2"),
+          eventsHostList = mutableListOf("3"),
+          friendsList = mutableListOf(),
+          profilePicUrl = "http://example.com/pic.jpg",
+          qrCodeUrl = "http://example.com/qr.jpg",
+          username = "john_doe")
+
+  @Before
+  fun setUp() {
+    messageRepository = MockMessageRepository()
+    userRepository = MockUserRepository()
+    viewModel = ChatViewModel(messageRepository, userRepository, opponentId)
+  }
+
+  @Test
+  fun `init and getMessages successful`() = runTest {
+    // Set up user repo
+
+    (userRepository as MockUserRepository).updateCurrentUserId("user1")
+    userRepository.addUser(opponent)
+
+    val msg1 = mockMessage.copy(sender = "user1", id = "msg1")
+    val msg2 = mockMessage.copy(sender = "user2", id = "msg2")
+    val messageHistory = messageRepository.createNewMessageHistory("user1", "user2")
+
+    messageRepository.addMessage(msg1, (messageHistory as Resource.Success).data)
+    messageRepository.addMessage(msg2, messageHistory.data)
+
+    viewModel =
+        ChatViewModel(messageRepository, userRepository, opponentId) // Initialize view model
+    viewModel.getMessages()
+
+    val expectedMessages = mutableListOf(msg1, msg2)
+    val uiState = viewModel.uiState.value
+    assert(uiState.userId == "user1")
+    assert(uiState.messagesLoadedFirstTime)
+    assert(expectedMessages == uiState.messageHistory.messages)
+    assert(uiState.opponentProfile == opponent)
+  }
+
+  @Test
+  fun `init getCurrentUserId failure and getMessages with null UserId`() = runTest {
+    mockkStatic(Log::class)
+    every { Log.d(any(), any()) } returns 0
+    (userRepository as MockUserRepository).updateCurrentUserId(null)
+
+    viewModel =
+        ChatViewModel(messageRepository, userRepository, opponentId) // Initialize view model
+
+    val uiState = viewModel.uiState.value
+    val userId = userRepository.getCurrentUserId()
+    assert(uiState.userId == null)
+    verify {
+      Log.d(
+          "ChatViewModel",
+          "Error getting user ID: ${(userId as Resource.Failure).throwable.message}")
+    }
+
+    viewModel.getMessages()
+
+    verify { Log.d("ChatViewModel", "Invalid state: User ID is null.") }
+    unmockkAll()
+  }
+
+  @Test
+  fun `initOpponent failure`() = runTest {
+    mockkStatic(Log::class)
+    every { Log.d(any(), any()) } returns 0
+    (userRepository as MockUserRepository).updateCurrentUserId("user1")
+
+    viewModel =
+        ChatViewModel(messageRepository, userRepository, opponentId) // Initialize view model
+
+    val uiState = viewModel.uiState.value
+    val opponentResource = userRepository.getUser(opponentId)
+    assert(uiState.opponentProfile == nullUser)
+    verify {
+      Log.d(
+          "ChatViewModel",
+          "Error getting opponent details: ${(opponentResource as Resource.Failure).throwable.message}")
+    }
+    unmockkAll()
```

Comment:
Might be easier to unmock after every test in one go by adding this line at the start of the code, before all the tests. 

`  @After fun testTeardown() = runTest { unmockkAll() }
`

Reply:
Since I don't use mockkStatic(...) in every test, I don't think it's necessary.